app/src/main/kotlin/com/wire/android/ui/login/Login.kt

```diff
@@ -0,0 +1,204 @@
+            }
+            LoginButton(modifier = Modifier.fillMaxWidth(), email = email.text, password = password.text)
+        }
+    }
+}
+
+@Composable
+private fun EmailInput(modifier: Modifier, email: TextFieldValue, onEmailChange: (TextFieldValue) -> Unit) {
+    WireTextField(
+        value = email,
+        onValueChange = onEmailChange,
+        placeholderText = stringResource(R.string.login_email_placeholder),
+        labelText = stringResource(R.string.login_email_label),
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Text,
+            imeAction = ImeAction.Next
+        ),
+        modifier = modifier,
+    )
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+private fun PasswordInput(modifier: Modifier, password: TextFieldValue, onPasswordChange: (TextFieldValue) -> Unit) {
+    val keyboardController = LocalSoftwareKeyboardController.current
+    var passwordVisibility by remember { mutableStateOf(false) }
+    WireTextField(
+        value = password,
+        onValueChange = onPasswordChange,
+        placeholderText = stringResource(R.string.login_password_placeholder),
+        labelText = stringResource(R.string.login_password_label),
+        keyboardOptions = KeyboardOptions(
+            keyboardType = KeyboardType.Password,
+            imeAction = ImeAction.Done
+        ),
+        keyboardActions = KeyboardActions(onDone = { keyboardController?.hide() }),
+        visualTransformation = if (passwordVisibility) VisualTransformation.None else PasswordVisualTransformation(),
+        trailingIcon = {
+            val image = if (passwordVisibility) Icons.Filled.Visibility else Icons.Filled.VisibilityOff
+            IconButton(onClick = { passwordVisibility = !passwordVisibility }) {
+                Icon(imageVector = image, "", modifier = Modifier.size(20.dp))
+            }
+        },
+        modifier = modifier,
+    )
+}
+
+@Composable
+private fun ForgotPasswordLabel(modifier: Modifier) {
+    Column(horizontalAlignment = Alignment.CenterHorizontally, modifier = modifier) {
+        val context = LocalContext.current
+        Text(
+            text = stringResource(R.string.login_forgot_password),
+            style = MaterialTheme.typography.body1.copy(
+                textDecoration = TextDecoration.Underline,
+                color = MaterialTheme.colors.primary
+            ),
+            textAlign = TextAlign.Center,
+            modifier = Modifier
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = null,
+                    onClick = { Toast.makeText(context, "Forgot password click 💥", Toast.LENGTH_SHORT).show() } //TODO
+                )
+        )
+    }
+}
+
+@OptIn(ExperimentalAnimationApi::class)
+@Composable
+private fun LoginButton(modifier: Modifier, email: String, password: String) {
+    var isLoading by remember { mutableStateOf(false) }
```

Comment:
shouldn't it be done on the viewmodel side ? I think you can just create a "mock" ViewModel and a "mock" State object that would hold stuff like that and glue the buisness logic part from Kalium later on into that ViewModel, but maybe you  just have a different approach and you do not use ViewModel at the moment for creating the UI elements

Reply:
Yes it can be in the ViewModel, current solution is temporary just to get the UI working, I assumed that this PR would only add UI, the logic and ViewModel would be added in the next